docs/design/assetgeneration.md

```diff
@@ -1,12 +1,12 @@
-# Installer asset generation based on dependency resolution
+# Asset generation
 
-## Goal
-
-Define generation of assets of varied types in installer based on a dependency graph.
+The installer internally uses a directed acyclic graph to represent all of the assets it creates as well as their dependencies. This process looks very similar to many build systems (e.g. Bazel, Make).
 
 ## Overview
 
-The installer generates assets based on the dependency [graph](./dependency.md). To generate an asset, all its dependencies have to be generated. The installer needs to define how assets declare this dependency and how it resolves the graph when asked to generated specific targets. Targets are chosen assets that need to be generated, the installer generates all assets that the target depends on and writes to disk only the targets, consuming any inputs from disk or state from a previous run.
+The installer generates assets based on the [dependency graph](#dependency-graph). Each asset seperately defines how it can be generated as well as its dependencies. Targets represent a set of assets that should be generated and written to disk for the user's consumption. When a user invokes the installer for a particular target, each of the assets in the set is generated as well as any dependencies. This eventually results in the user being prompted for any missing information (e.g. administrator password, target platform).
+
+The installer is also able to read assets from disk if they have been provided by the user. In the event that an asset exists on disk, the install won't generate the asset, but will instead consume the asset from disk (removing the file). This allows the installer to be run multiple times, using the assets generated by the previous invocation. It also allows a user to make modifications to the generated assets before continuing to the next target.
```

Comment:
consume?  Why?  Personally I really like the "git-ops" model where one stores configuration files in git and code reacts to commits.  I'd expect people want to store their install configs in git and not have to undo the installer deleting things.

Reply:
> consume? Why?...

+1 to this, but the installer doesn't do any disk-reading now, so we can revisit when we add that.